### PR TITLE
Feature/cluster logic

### PR DESCRIPTION
**Pull request by nmenshov**

### CODE REVIEW

File: proto/bob.proto [hunk 1]
```diff
@@ -46,8 +46,9 @@ message OpStatus {
 
 // Put operation options
 message PutOptions {
-   bool force_node = 1; // Force operation to be served by node to which it comes
-   bool overwrite = 2; // Overwrite data in case of id existens
+    repeated string remote_nodes = 1;
```
Comment (agend) [inline proto/bob.proto]: what is format for this field?

File: src/core/backend/core.rs [hunk 2]
```diff
@@ -16,7 +17,8 @@ use std::{pin::Pin, sync::Arc};
 pub struct BackendOperation {
     vdisk_id: VDiskId,
     disk_path: Option<DiskPath>,
-    alien: bool, // flag marks data belonging for different node
+    remote_node_name: Option<String>, // save data to alien/<remote_node_name>
```
Comment (agend) [inline src/core/backend/core.rs]: if we have `remote_node_name` do we need `alien` flag?

File: src/core/mapper.rs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            local_node_name: config.name.as_ref().unwrap().to_string(),
+            disks: config
+                .disks()
+                .iter()
+                .map(|d| DiskPath::new(&d.name.clone(), &d.path.clone()))
+                .collect(),
+            nodes,
+        }
+    }
+    pub fn new2(
```
Comment (agend) [inline src/core/mapper.rs]: what is `new2`??

File: src/core/mapper.rs [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+}
+
+impl VDiskMapper {
+    pub fn new(vdisks: Vec<VDisk>, config: &NodeConfig, cluster: &ClusterConfig) -> VDiskMapper {
+        let (nodes, vdisks) = Self::prepare_nodes(vdisks, &cluster);
+
+        VDiskMapper {
+            vdisks,
+            local_node_name: config.name.as_ref().unwrap().to_string(),
+            disks: config
```
Comment (agend) [inline src/core/mapper.rs]: Could we call `new2` here with `new2(..., config.disks...)`?

File: src/core/backend/core.rs (hunk 2 above)
Comment (nmenshov) [reply src/core/backend/core.rs]: in next branch i want to refact this logic

File: proto/bob.proto (hunk 1 above)
Comment (nmenshov) [reply proto/bob.proto]: it's array of node names

File: src/core/cluster/quorum.rs [hunk 5]
```diff
@@ -0,0 +1,1007 @@
...
+                        err
+                    )))
+                }
+            }
+        };
+        Put(p.boxed())
+    }
+
+    //todo check no data (no error)
+    fn get_clustered(&self, key: BobKey) -> Get {
```
Comment (agend) [inline src/core/cluster/quorum.rs]: Add async suffix for function to mark as it's async

File: src/core/mapper.rs (hunk 3 above)
Comment (nmenshov) [reply src/core/mapper.rs]: rename) it used when you run cluster from script and must run node with target name (not from node.yaml config)

**Pull request merged**